src/cpp/ext/otel/otel_client_call_tracer.cc

```diff
@@ -374,12 +374,12 @@ OpenTelemetryPluginImpl::ClientCallTracer::StartNewAttempt(
     if (transparent_retries_ != 0 || retries_ != 0) {
       is_first_attempt = false;
     }
-    attempt_num = retries_;
     if (is_transparent_retry) {
       ++transparent_retries_;
     } else {
       ++retries_;
     }
+    attempt_num = retries_ - 1;  // Sequence starts at 0
```

Comment:
Why not go back to setting this before we increment `retries_`, so that we don't need to subtract again?

Reply:
consider the case where we transparently retry an RPC. What we want is -
```
previous-rpc-attempts= 0, transparent-retry=false
previous-rpc-attempts=0, transparent-retry=true 
```

The previous code is incorrect since it would increment `retries_` on the first attempt, resulting in an incorrect -
```
previous-rpc-attempts= 0, transparent-retry=false
previous-rpc-attempts=1, transparent-retry=true 
```